### PR TITLE
Feature/orb 1300 add policy status agent list

### PR DESCRIPTION
**Pull request by mclcavalcante**
Adds the following map structure field to response of agent list endpoint, in which the key is the policy ID and the value is the policy information. All retrieve from lastHBData on DB

``
   "policy_status": {
      "43230fb0-9b28-d73c-0d40-efc8446bcfe8":{
          "name": "test",
          "status": "running"
      }
   }
``
image

### CODE REVIEW

**Review by lpegoraro — APPROVED** [review]
LGTM

**Comment by lpegoraro** [comment]
@mclcavalcante Add a print of it running locally in the Network tab in the Web Console of the Browser or from Postman, that ensures and it is easier on the code reviewer

File: fleet/api/http/endpoint.go [hunk 1]
```diff
@@ -11,6 +11,7 @@ package http
 import (
 	"context"
 	"github.com/go-kit/kit/endpoint"
+	"github.com/mitchellh/mapstructure"
```
Comment (weyrick) [inline fleet/api/http/endpoint.go:14]: why are we using this instead of JSON decode?

**Review by weyrick — CHANGES_REQUESTED** [review]

File: fleet/api/http/endpoint.go [hunk 2]
```diff
@@ -290,16 +291,33 @@ func listAgentsEndpoint(svc fleet.Service) endpoint.Endpoint {
 			},
 			Agents: []agentRes{},
 		}
+
 		for _, ag := range page.Agents {
+			policyStatus := make(map[string]interface{})
+			var hb fleet.Heartbeat
+			err = mapstructure.Decode(ag.LastHBData, &hb)
+			if err != nil {
+				return nil, err
```
[the remark this answers is not in the document]
Comment (lpegoraro) [reply fleet/api/http/endpoint.go:300]: We will extract the method into the service that's better either way

File: fleet/agent_service.go [hunk 3]
```diff
@@ -273,3 +274,25 @@ func (svc fleetService) ViewAgentInfoByChannelIDInternal(ctx context.Context, ch
 	}
 	return res, nil
 }
+
+func (svc fleetService) GetPoliciesState(ctx context.Context, agent Agent) (map[string]interface{}, error) {
+
+	jsonHb, err := json.Marshal(agent.LastHBData)
+	if err != nil {
+		svc.logger.Error("failed to marshal heartbeat data", zap.Error(err))
```
Comment (weyrick) [inline fleet/agent_service.go:282]: need to return here on error so it doesn't continue

File: fleet/agent_service.go [hunk 4]
```diff
@@ -273,3 +274,25 @@ func (svc fleetService) ViewAgentInfoByChannelIDInternal(ctx context.Context, ch
...
+	}
+	var hb Heartbeat
+	if err = json.Unmarshal(jsonHb, &hb); err != nil {
+		svc.logger.Error("failed to unmarshal heartbeat data", zap.Error(err))
```
Comment (weyrick) [inline fleet/agent_service.go:286]: same here

**Review by weyrick — CHANGES_REQUESTED** [review]

**Review by weyrick — APPROVED** [review]

**Pull request merged**